### ISSUE TITLE
Suggestion: Export App

### ISSUE THREAD

**Issue by MeneDev** [caprover/caprover#796]
**Is your feature request related to a problem? Please describe.**
I have created some apps in an "ad-hoc" way by using the very handy "Deploy via ImageName" feature. I would however like to have the result in a reproducible format.

**Describe the solution you'd like**
Somewhere in the UI for the App I'd like an export option. As a minimal viable solution I see "export as captain-definition", which would allow me to download the captain-definition as file. A more complex solution would be "export to git".

**Comment by githubsaturn** [comment]
I like the general idea. The app could export into a Docker compose / CapRover one click app format with all volumes, ports and etc, and it's easy to read for humans.

My very big concern is that this only works if the app is completely vanilla. And that will cause lots of confusion and problems for the end user. The exported app won't work if any of the following items are true:
- If it's using a locally built image
- If it's relying on a local volume attach to it
- If it's relying on other apps

**Comment by MeneDev** [comment]
You're right, it has a lot of edge-cases. Maybe not worth it and causes more frustration than help. Maybe listing those limitations next to the export button is enough? Cannot really judge your user base.

**Comment by maietta** [comment]
Because Caprover is simply a deployment and management front-end for a self-contained Nginx ingress proxy for docker services, I feel as if the Docker API or command line interface itself would be the best interface for what you are wanting to accomplish. For example, you can use the "docker service ls" command to list your services, then inspect the one you are after. With this you can know which networks, volumes etc are used for that service. The best part is that you'd have something that runs independently of Caprover and is totally portable for any docker swarm environment.

https://docs.docker.com/engine/reference/commandline/service_inspect/

I've thought about this myself, but not as a built-in set of functions for Caprover as I see Caprover as a deployment-only tool. A tool like this would be a fairly tall order. Just my two cents.

**Comment by GiuseppeRaso** [comment]
I am coming from #980 because I'd like to see a button to duplicate an app. I think "duplicate app" could be much simpler to implement than "export app" because if you duplicate, the duplicated app remains into the same instance of caprover, so it can use things that are local like other apps, built images etc... Probably, but maybe I'm wrong, it wouldn't even need to use docker-compose/one-click-apps syntax. You also should be prompted to change name and persistance labels/folders.
Do you think it's possible in theory or there would be some problems?

**Comment by perzeuss** [comment]
I think, in theory, it would be possible by doing the following

1. create a new app
2. copy over the environment values (optional? checkbox?)
3. copy over the deployment settings, if present (e.g. git settings) 
4. start the app with the built caprover image which was built for the original app


Of course, when the app relies on other apps (or if envs havn't been copied), there are manual steps to do. But I think it would be absolutely fine to leave this responsibility with the user ..
